network/conduit.go

```diff
@@ -3,57 +3,23 @@
 package network
 
 import (
+	"context"
 	"errors"
 	"fmt"
-	"sort"
 
 	"github.com/onflow/flow-go/model/flow"
 )
 
-// Channel specifies a virtual and isolated communication medium.
-// Nodes subscribed to the same channel can disseminate epidemic messages among
-// each other, i.e.. multicast and publish.
-type Channel string
-type ChannelList []Channel
+// ConduitFactory is an interface type that is utilized by the Network to create conduits for the channels.
+type ConduitFactory interface {
+	// RegisterAdapter sets the Adapter component of the factory.
+	// The Adapter is a wrapper around the Network layer that only exposes the set of methods
+	// that are needed by a conduit.
+	RegisterAdapter(Adapter) error
 
-func (c Channel) String() string {
-	return string(c)
-}
-
-// Len returns length of the ChannelList in the number of stored Channels.
-// It satisfies the sort.Interface making the ChannelList sortable.
-func (cl ChannelList) Len() int {
-	return len(cl)
-}
-
-// Less returns true if element i in the ChannelList  is less than j based on the numerical value of its Channel.
-// Otherwise it returns true.
-// It satisfies the sort.Interface making the ChannelList sortable.
-func (cl ChannelList) Less(i, j int) bool {
-	return cl[i] < cl[j]
-}
-
-// Swap swaps the element i and j in the ChannelList.
-// It satisfies the sort.Interface making the ChannelList sortable.
-func (cl ChannelList) Swap(i, j int) {
-	cl[i], cl[j] = cl[j], cl[i]
-}
-
-// ID returns hash of the content of ChannelList. It first sorts the ChannelList and then takes its
-// hash value.
-func (cl ChannelList) ID() flow.Identifier {
-	sort.Sort(cl)
-	return flow.MakeID(cl)
-}
-
-// Contains retuns true if the ChannelList contains the given channel.
-func (cl ChannelList) Contains(channel Channel) bool {
-	for _, c := range cl {
-		if c == channel {
-			return true
-		}
-	}
-	return false
+	// NewConduit creates a conduit on the specified channel.
+	// Prior to creating any conduit, the factory requires an Adapter to be registered with it.
+	NewConduit(context.Context, Channel) (Conduit, error)
```

Comment:
can you clarify why we're specifying a `Channel` for the `Conduit`? Is it to allow specifying different channels depending on if it's a normal or a test channel?

Reply:
We resolved this in our meeting.